src/main/java/org/candlepin/subscriptions/inventory/db/model/InventoryHostFacts.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.inventory.db.model;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.StringUtils;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Represents an inventory host's facts.
+ */
+public class InventoryHostFacts {
+    private static final Logger log = LoggerFactory.getLogger(InventoryHostFacts.class);
+
+    private String account;
+    private String displayName;
+    private String orgId;
+    private int cores;
+    private int sockets;
+    private boolean isRhel;
+    private String syncTimestamp;
+    private Set<String> products;
+
+    public InventoryHostFacts(String account, String displayName, String orgId, String cores, String sockets,
```

Comment:
SonarQube flagged this constructor for having too many arguments. I have to agree, especially given that they are all `String`. Let's remove this constructor, and use the setters instead. We can add a few overloaded setters that take `String` where we want to use `asInt`, `asBoolean`, etc.

Reply:
I don't know that we can do that.  The object is created in `InventoryHost` via a `@SqlResultSetMapping` annotation.  That annotation also allows for mapping to an entity via field injection, but `InventoryHostFacts` isn't an entity.  It's just a convenience class we use as a grab bag.

Reply:
Ack.